Sources/TuistKit/Utils/ImportSourceCodeScanner.swift

```diff
@@ -5,65 +5,84 @@ enum ProgrammingLanguage {
 
-        let matches = regex.matches(
-            in: code,
-            options: [],
-            range: NSRange(
-                location: 0,
-                length: code.utf16.count
-            )
-        )
+        let lines = codeWithoutComments.components(separatedBy: .newlines)
 
-        for match in matches {
-            if let moduleRange = Range(match.range(at: 1), in: code) {
-                let module = String(code[moduleRange])
-                let topLevelModule = module.split(separator: ".").first.map(String.init)
-                if let topLevelModule, !modules.contains(topLevelModule) {
-                    modules.append(topLevelModule)
+        for line in lines {
+            let range = NSRange(location: 0, length: line.utf16.count)
+            let matches = regex.matches(in: line, options: [], range: range)
+
+            for match in matches {
+                let match = switch language {
+                case .swift:
+                    processMatchSwift(match: match, line: line)
+                case .objc:
+                    processMatchObjc(match: match, line: line)
+                }
+                if let match {
+                    if !result.contains(where: { $0 == match.module }) {
+                        result.append(match.module)
+                    }
                 }
             }
         }
-
-        return modules
+        return result
     }
 
-    func extractAllImportsObjc(from text: String) throws -> [String] {
-        let pattern = "@import\\s+([A-Za-z_0-9]+)|#(?:import|include)\\s+<([A-Za-z_0-9-]+)/"
-
-        let regex = try NSRegularExpression(pattern: pattern, options: [])
-        let matches = regex.matches(
-            in: text,
-            options: [],
-            range: NSRange(
-                location: 0,
-                length: text.utf16.count
+    private func processMatchSwift(match: NSTextCheckingResult, line: String) -> Match? {
+        var module: Match?
+        if let moduleRange = Range(match.range(at: 1), in: line),
+           let foundModule = String(line[moduleRange]).split(separator: ".").first.map(String.init)
+        {
+            module = Match(
+                module: foundModule,
+                range: moduleRange
             )
-        )
+        }
+        return module
+    }
 
-        return matches.compactMap { match in
-            if let range = Range(match.range(at: 1), in: text) {
-                return String(text[range])
-            } else if let range = Range(match.range(at: 2), in: text) {
-                return String(text[range])
-            }
-            return nil
+    func processMatchObjc(match: NSTextCheckingResult, line: String) -> Match? {
+        var result: Match?
+        if let range = Range(match.range(at: 1), in: line) {
+            result = Match(
+                module: String(line[range]),
+                range: range
+            )
```

Comment:
```suggestion
            return Match(
                module: String(line[range]),
                range: range
            )
```

We can return directly instead of saving the match into a variable first